tests/unit/bcss_notify_callback/test_lambda_function.py

```diff
@@ -0,0 +1,260 @@
+import hashlib
+import hmac
+import logging
+import pytest
+import requests
+import lambda_function as lf
+from unittest.mock import MagicMock
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def setup(monkeypatch):
+    monkeypatch.setenv("APPLICATION_ID", "application_id")
+    monkeypatch.setenv("NOTIFY_API_KEY", "api_key")
+
+
+def test_generate_hmac_signature_valid(setup):
+    """Test that an HMAC SHA-256 signature is generated correctly."""
+    body = "body"
+    signature = hmac.new(
+        "application_id.api_key".encode(), body.encode(), hashlib.sha256
+    ).hexdigest()
+
+    assert lf.generate_hmac_signature("application_id.api_key", body) == signature
+
+
+def test_validate_signature_valid(setup):
+    """Test that a valid signature passes verification."""
+    body = "body"
+    signature = hmac.new(
+        "application_id.api_key".encode(), body.encode(), hashlib.sha256
+    ).hexdigest()
+
+    assert lf.validate_signature(signature, "application_id.api_key", body)
+
+
+def test_validate_signature_invalid(setup):
+    """Test that an invalid signature does not pass verification."""
+    body = "body"
+    signature = hmac.new(
+        "application_id.api_key".encode(), body.encode(), hashlib.sha256
+    ).hexdigest()
+
+    assert not lf.validate_signature(signature, "test", body)
+
+
+# There is a lot of code in this handler, worth breaking down into smaller chunks?
```

Comment:
Good you are thinking about this and not attempting to test the whole thing. We will break this down into smaller functions and then the tests will be much easier to write.